src/Queue.php

```diff
@@ -0,0 +1,302 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace AvtoDev\AmqpRabbitLaravelQueue;
+
+use DateTime;
+use RuntimeException;
+use Illuminate\Support\Str;
+use Interop\Amqp\AmqpQueue;
+use Illuminate\Container\Container;
+use Interop\Amqp\AmqpMessage as Message;
+use Enqueue\AmqpExt\AmqpContext as Context;
+use Enqueue\AmqpExt\AmqpConsumer as Consumer;
+use Enqueue\AmqpExt\AmqpProducer as Producer;
+use Illuminate\Contracts\Debug\ExceptionHandler;
+use Illuminate\Contracts\Queue\Queue as QueueContract;
+use Enqueue\AmqpTools\RabbitMqDelayPluginDelayStrategy;
+use Interop\Queue\Exception\DeliveryDelayNotSupportedException;
+
+class Queue extends \Illuminate\Queue\Queue implements QueueContract
+{
+    /**
+     * @var AmqpQueue
+     */
+    protected $queue;
+
+    /**
+     * @var int
+     */
+    protected $timeout;
+
+    /**
+     * @var Context
+     */
+    protected $connection;
+
+    /**
+     * Queue constructor.
+     *
+     * @param Container $container
+     * @param Context   $connection
+     * @param AmqpQueue $queue
+     * @param int       $timeout    Timeout in milliseconds
+     */
+    public function __construct(Container $container, Context $connection, AmqpQueue $queue, int $timeout = 0)
+    {
+        $this->container  = $container;
+        $this->connection = $connection;
+        $this->queue      = $queue;
+        $this->timeout    = \max(0, $timeout);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param int|null $sleep Sleep for a some time before broker calling, in micro seconds
+     */
+    public function size($queue = null, ?int $sleep = 2000): int
+    {
+        if (\is_int($sleep)) {
+            \usleep($sleep); // Required for broker (for calling in a loop)
+        }
+
+        return $this->connection->declareQueue($this->queue);
+    }
+
+    /**
+     * Push a new job onto the queue.
+     *
+     * @param object|string $job
+     * @param mixed         $data
+     * @param string        $queue
+     * @param int|null      $priority Message priority
+     *
+     * @return void
+     */
+    public function push($job, $data = '', $queue = null, ?int $priority = null): void
+    {
+        $options = [];
+
+        if (\is_object($job) && $job instanceof PrioritizedJobInterface) {
```

Comment:
Same extracting of priority as in `later()`. May be make private method with check `$job` and `$priority`.